Processor/processor.py

```diff
@@ -18,6 +21,33 @@ def __init__(self, soup, section, base_url, carousels):
         self.curr_text = ''
         self.is_curr_tag_headline = False
         self.is_curr_tag_sub_headline = False
+        
+    @staticmethod
+    def suffix_resize_url(url):
+        """Inserts '-resize' into a URL before the file extension."""
+        parsed_url = urllib.parse.urlparse(url)
+        # Handle cases where there might be query parameters or fragments
+        path_parts = parsed_url.path.rsplit(".", 1)  # Split only on the last "."
+        if len(path_parts) == 2:
+            new_path = f"{path_parts[0]}-resize.{path_parts[1]}"
+        else:
+            # If no extension, simply append to the end
+            new_path = parsed_url.path + "-resize"
+        # Reconstruct the URL with the modified path
+        return parsed_url._replace(path=new_path).geturl()
+
+    @staticmethod
+    def resize_long_image(url):
```

Comment:
shouldn't this method reside in common_utility.py ? (you can pass URL or Img object and desired height & width)
Actual rescaling launch is static method.

Reply:
Took care in https://github.com/google/html-to-jsonld-converter/pull/4

thanks